### PR TITLE
Tidy-up noexcept move operations in basic_fields

### PR DESCRIPTION
**Pull request by djarek**
Adds failing tests for #894.

### CODE REVIEW

**Comment by vinniefalco** [comment]
Please squash this down into one commit, suggested message "Tidy up basic_fields exception specifiers"

**Comment by vinniefalco** [comment]
@glenfe how do you feel about these changes?

File: include/boost/beast/http/fields.hpp [hunk 1]
```diff
@@ -245,7 +258,8 @@ class basic_fields
         The state of the moved-from object is
         as if constructed using the same allocator.
     */
-    basic_fields& operator=(basic_fields&&);
+    basic_fields& operator=(basic_fields&&) noexcept(
+        alloc_traits::propagate_on_container_move_assignment::value);
```
Comment (glenfe) [inline include/boost/beast/http/fields.hpp]: Should also require `is_nothrow_move_assignable<Allocator>::value` (in addition to POCMA).

Comment (djarek) [reply include/boost/beast/http/fields.hpp]: This is part of the requirements of the Allocator concept. Should we check them with static_asserts?

Comment (glenfe) [reply include/boost/beast/http/fields.hpp]: For the latter, if in C++17 or higher mode, it should be in the noexcept condition. i.e. `noexcept(alloc_traits::propagate_on_container_move_assignment::value || alloc_traits::is_always_equal::value)`.

Before C++17 the Allocator requirements were a little more relaxed. You could use `noexcept(alloc_traits::propagate_on_container_move_assignment::value && is_nothrow_move_assignable<Allocator>::value)` if in C++11/14 mode, but in practice I think you're fine with what you have (and just `static_assert`).

**Comment by vinniefalco** [comment]
Commit message should read:
```
Tidy up basic_fields exception specifiers

fix #894

Signed-off-by: Damian Jarek <damian.jarek93@gmail.com>
```

Note the blank lines. Also, please add a line to CHANGELOG.md (rebase against my **147** branch first)

Thanks!

**Comment by vinniefalco** [comment]
Merged to **147**, thanks!

**Pull request closed without merging**